### PR TITLE
Feat/multiple switches

### PR DESCRIPTION
**Pull request by f-erdi1990**
As discussed  - tests now cover the most basic functionality, but very inefficient. On next rotation, i will revert to your original method of having precomposed elements to plug into each unit test and have less repetition in code.

### CODE REVIEW

File: contracts/DeadMansSwitch.sol [hunk 1]
```diff
@@ -6,23 +6,93 @@ import "hardhat/console.sol";
 error DeadMansSwitchError();
 
 contract DeadMansSwitch {
-    bool public isAlive;
+    uint public counter;
```
Comment (arein) [inline contracts/DeadMansSwitch.sol:9]: consider using https://docs.openzeppelin.com/contracts/3.x/api/utils#Counters

File: contracts/DeadMansSwitch.sol [hunk 2]
```diff
@@ -6,23 +6,93 @@ import "hardhat/console.sol";
...
+        Switch memory mySwitch = Switch({
+            onlyOwner:_onlyOwner,
+            cadence:_cadence,
+            downCadence:_downCadence,
+            startWindowBlock:(block.number + _downCadence),
+            endWindowBlock:(block.number + _cadence)
+            }
+        );
+        uint switchId = _getId();
+        require(!_exists(switchId), "Dead Man's Switch: Switch already minted");
```
Comment (arein) [inline contracts/DeadMansSwitch.sol:52]: consider also requiring only one switch per address

File: contracts/DeadMansSwitch.sol [hunk 3]
```diff
@@ -6,23 +6,93 @@ import "hardhat/console.sol";
...
+        switches[switchId] = mySwitch;
+        console.log("Successful mint of switch number:", switchId);
```
Comment (arein) [inline contracts/DeadMansSwitch.sol:54]:
```suggestion
        console.log("Successful mint of switch id ", switchId);
```

File: contracts/DeadMansSwitch.sol [hunk 4]
```diff
@@ -6,23 +6,93 @@ import "hardhat/console.sol";
...
+    }
+
+    function _exists(uint _switchId) internal view returns (bool) {
+        if (switches[_switchId].onlyOwner != address(0)){
+            return true;
+        }
+        return false;
+    }
+
+    function heartbeat(uint _switchId) external returns (bool) {
```
Comment (arein) [inline contracts/DeadMansSwitch.sol:66]:
```suggestion
    function heartbeat(uint _switchId) external onlyOwner returns (bool) {
```

**Review by arein — APPROVED** [review]
PR needs to be rebased

we need public a way to see all switches - can be a dedicated PR

File: contracts/DeadMansSwitch.sol (hunk 2 above)
Comment (f-erdi1990) [reply contracts/DeadMansSwitch.sol:52]: maybe for this initial contract, but technically it should be possible, right?

**Pull request merged**